### DISCUSSION TITLE
Does warpX have a constant source capability?

### DISCUSSION THREAD

**Discussion by moisesa2** [BLAST-WarpX/warpx#4085] [Q&A]
I was hoping to run simple plasma discharge cases without the use of the MCC module and I was wondering if the above is built in already. The documentation mentions continuous_injection but It's not clear if that is what I am looking for.

**Comment by dpgrote** [comment]
@moisesa2 Presumably you mean continuously adding plasma over some spatial region (like the replenishment feature in my sheath branch). WarpX currently does not have this ability in the C++ version. The `do_continuous_injection` is for when there is a moving window and adds plasma in the new space opened up as the grid moves.
This feature adding plasma can be done in the Python version. You would write a function that creates the new particles and passed them to the `add_particles` routine. Then that function would be installed to be called as a `particleinjection` callback.
